Motion/app/src/main/java/com/example/android/motion/demo/containertransform/CheeseArticleFragment.kt

```diff
@@ -111,5 +118,45 @@ class CheeseArticleFragment : Fragment() {
         toolbar.setNavigationOnClickListener { v ->
             v.findNavController().popBackStack()
         }
+
+        requireActivity().onBackPressedDispatcher.addCallback(
+            viewLifecycleOwner,
+            object : OnBackPressedCallback(true) {
+                override fun handleOnBackPressed() {
+                    // This invokes the sharedElementReturnTransition, which is
+                    // MaterialContainerTransform.
+                    // TODO: It pops the scaleX/Y back to 1f. It would look nicer to retain the
+                    //       scale. How should we handle it?
+                    findNavController().popBackStack()
+                }
+
+                override fun handleOnBackProgressed(backEvent: BackEventCompat) {
+                    val progress = backEvent.progress
+                    // Translate as far as 20% of finger movement.
+                    val translation = progress * background.width * 0.2f
```

Comment:
the formula for translationX is progress * maxXShift, where maxXShift is windowWidth / 20

e.g. https://github.com/android/platform-samples/tree/main/samples/user-interface/predictiveback#custom-progress-api-animation

Reply:
Thank you. I didn't know we had a motion spec.